src/reconstruct/helper.ts

```diff
@@ -1,20 +1,69 @@
-import { NodeId, ParentInformation, RNode, RType } from '../r-bridge';
-import { ReconstructionConfiguration } from './reconstruct';
+import { NodeId, ParentInformation, RNode, RType } from '../r-bridge'
+import { ReconstructionConfiguration } from './reconstruct'
 
 
 /*
 --helper function--
 */
 export type Selection = Set<NodeId>
+interface PrettyPrintLinePart {
+	part:   string
+	spaces: number
+}
 export interface PrettyPrintLine {
-	line: string
+	line:   string
 	indent: number
 }
-export function plain(text: string): PrettyPrintLine[] {
-	return [{ line: text, indent: 0 }]
+export interface WrappedPrettyPrintLine { //change name to be more fitting
+	linePart: PrettyPrintLinePart[]
+	indent:   number
+}
+export function plain(text: string,space: number): Code {
+	return [unwrapPrettyPrintLine({linePart: [{part: text,spaces: space}],indent: 0})]
 }
 export type Code = PrettyPrintLine[]
 
+export function unwrapPrettyPrintLine(wrappedLine: WrappedPrettyPrintLine): PrettyPrintLine {
+	const text = ''
+	for(let index = 0; index < wrappedLine.linePart.length; index++) {
+		const element = wrappedLine.linePart[index]
```

Comment:
As you are never writing to the respective line part, you may be better of replacing these two lines with

```suggestion
	for(const element of wrappedLine.linePart) {
```